### PR TITLE
Add field option API suite

### PR DESCRIPTION
**Pull request by rocketspacer**
Add fieldOption APIs base on [jira docs](https://docs.atlassian.com/jira/REST/cloud/)

P/s: How often do you push new versions to npm?

### CODE REVIEW

**Comment by rocketspacer** [comment]
@KanoYugoro Hi

File: src/jira.js [hunk 1]
```diff
@@ -696,6 +712,82 @@ export default class JiraApi {
...
+   */
+  listFieldOptions(fieldKey) {
+    return this.doRequest(this.makeRequestHeader(this.makeUri({
+      pathname: `/field/${fieldKey}/option`,
+    })));
+  }
+
+  /** Creates or updates an option for a select list issue field.
+   * [Jira Doc](http://docs.atlassian.com/jira/REST/latest/#api/2/field/{fieldKey}/option-putOption)
+   * @name upsertFieldOption
```
Comment (KanoYugoro) [inline src/jira.js]: I think that this function could use a better name.

**Comment by KanoYugoro** [comment]
I update this package whenever I have changes that I feel need to come in.  And whenever I need to release a new version, since the release process is trivial for me, I just release a new version.

File: src/jira.js [hunk 2]
```diff
@@ -696,6 +712,82 @@ export default class JiraApi {
...
+   */
+  deleteFieldOption(fieldKey, optionId) {
+    return this.doRequest(this.makeRequestHeader(this.makeUri({
+      pathname: `/field/${fieldKey}/option/${optionId}`,
+    }), {
+      method: 'DELETE',
+      followAllRedirects: true,
+    }));
+  }
+
```
Comment (KanoYugoro) [inline src/jira.js]: I also don't see any unit tests.  At least add some tests to ensure these functions are hitting the expected URL's.

File: src/jira.js (hunk 1 above)
Comment (rocketspacer) [inline src/jira.js]: Can you name a few?

Comment (KanoYugoro) [inline src/jira.js]: I guess it's fine, I was being overly picky.

**Comment by rocketspacer** [comment]
@KanoYugoro I've added the unit tests. Do you want me to refactor all other unit tests as API Suites?  By the way, do you work for Atlassian?

**Comment by KanoYugoro** [comment]
I do not work for atlassian.  You're welcome to do whatever refactors you'd like to do, but I'd do them in a different PR.  This one looks fine as is.

File: test/jira-tests.js [hunk 3]
```diff
@@ -539,9 +539,45 @@ describe('Jira API Tests', () => {
       result.should.eql('http://jira.somehost.com:8080/rest/api/2.0/component/someComponentNumber');
     });
 
-    it('listFields hits proper url', async () => {
```
Comment (KanoYugoro) [inline test/jira-tests.js]: Looks like you may have accidentally removed this unit test?  The method still exists in the main file.

Comment (rocketspacer) [inline test/jira-tests.js]: No, I didn't remove it. I grouped it under "Field APIs Suite Tests" along with createCustomFields as described in [Jira standard ](https://docs.atlassian.com/jira/REST/cloud/#api/2/field)

Comment (KanoYugoro) [inline test/jira-tests.js]: Ah, I see.  My apologies for missing that.  I'm fine bringing this in now

**Pull request merged**